### PR TITLE
Added support to read device info from MSAL

### PR DESCRIPTION
**Pull request by oldalton**
This PR implements MSAL getDeviceInfo public API that allows retrieving whether device is in the shared mode or not. In the future, we can return more information on the device information object.

### CODE REVIEW

File: MSAL/src/instance/MSALAccountsProvider.m [hunk 1]
```diff
@@ -441,41 +440,4 @@ - (BOOL)setCurrentPrincipalAccountId:(MSIDAccountIdentifier *)currentAccountId e
...
-    }
-    
-    return NO;
-}
-
-- (MSIDSSOExtensionGetAccountsRequest *)copyAndClearCurrentSSOExtensionRequest API_AVAILABLE(ios(13.0), macos(10.15))
-{
-    @synchronized (self)
-    {
-        if (!self.currentRequest)
```
Comment (jasoncoolmax) [inline MSAL/src/instance/MSALAccountsProvider.m]: `currentRequest` is not a static variable, how can it prevent multiple `allAccountsFromDevice` calls from running at the same time?

Another quesiton is: why do we only allow one `allAccountsFromDevice` call at a time? :)

File: MSAL/src/MSALPublicClientApplication.m [hunk 2]
```diff
@@ -1332,6 +1322,56 @@ - (void)signoutWithAccount:(nonnull MSALAccount *)account
...
+    NSError *requestParamsError;
+    MSIDRequestParameters *requestParams = [self defaultRequestParametersWithError:&requestParamsError];
+    
+    if (!requestParams)
+    {
+        block(nil, requestParamsError);
+        return;
+    }
+    
+    [request deviceInfoWithRequestParameters:requestParams completionBlock:block];
```
Comment (antrix1989) [inline MSAL/src/MSALPublicClientApplication.m]: `request` is a local variable, can it be deallocated before we get a result in completion block?

File: MSAL/src/instance/MSALAccountsProvider.m (hunk 1 above)
Comment (oldalton) [reply MSAL/src/instance/MSALAccountsProvider.m]: @jasoncoolmax, just an optimization. I didn't see a reason why would developer need multiple requests for deviceInfo for the same public client app, but I might be wrong. I thought to have one instance of deviceInfo provider per public client application (need to change current code a bit), and not enforce it globally across process. Not feeling very strong neither ways, so if you have other suggestions, more than happy to adopt them :)

File: MSAL/src/MSALPublicClientApplication.m (hunk 2 above)
Comment (oldalton) [reply MSAL/src/MSALPublicClientApplication.m]: It captures self internally, so won't be deallocated. 
I was actually thinking to make it a property on public client app, so we can have one device info request at a time for one public client app. However, I think we can implement it in many different ways, so let me know if you have any recommendations here.

**Comment by jasoncoolmax** [comment]
I think this PR looks good to me, except that we need to decide if we want to error out if developers call multiple `getDeviceInformationWithParameters` using the same PCA.

**Comment by oldalton** [comment]
@jasoncoolmax, I changed it to not return error right now (every call to getDeviceInfo API will create new provider). Let's have a discussion next week whether we want to keep this behavior and we can make a change as a separate PR if needed.

**Pull request merged**